multinet/api/tests/test_workspace.py

```diff
@@ -51,48 +70,281 @@ def test_workspace_rest_create(authenticated_api_client: APIClient):
+
+    r = api_client.delete(f'/api/workspaces/{workspace.name}/')
 
     assert r.status_code == 401
 
     # Assert relevant objects are not deleted
-    assert Workspace.objects.filter(name=owned_workspace.name).exists()
-    assert arango_system_db().has_database(owned_workspace.arango_db_name)
+    assert Workspace.objects.filter(name=workspace.name).exists()
+    assert arango_system_db().has_database(workspace.arango_db_name)
 
 
 @pytest.mark.django_db
 def test_workspace_rest_delete_forbidden(
-    workspace_factory: WorkspaceFactory, authenticated_api_client: APIClient
+    workspace: Workspace, user: User, authenticated_api_client: APIClient
 ):
-    # Create workspace this way, so the authenticated user isn't an owner
-    workspace: Workspace = workspace_factory()
+    workspace.set_readers([user])
+    response = authenticated_api_client.delete(f'/api/workspaces/{workspace.name}/')
+    assert response.status_code == 403
+
+    # Assert relevant objects are not deleted
+    assert Workspace.objects.filter(name=workspace.name).exists()
+    assert arango_system_db().has_database(workspace.arango_db_name)
+
+
+@pytest.mark.django_db
+def test_workspace_rest_delete_no_access(
+    private_workspace_factory: PrivateWorkspaceFactory, authenticated_api_client: APIClient
+):
+    # Create workspace this way, so the authenticated user has no access
+    workspace: Workspace = private_workspace_factory()
     r = authenticated_api_client.delete(f'/api/workspaces/{workspace.name}/')
 
-    assert r.status_code == 403
+    # Without any access, don't reveal that this workspace exists at all
+    assert r.status_code == 404
 
     # Assert relevant objects are not deleted
     assert Workspace.objects.filter(name=workspace.name).exists()
     assert arango_system_db().has_database(workspace.arango_db_name)
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize('permission', [WorkspacePermission.owner, WorkspacePermission.maintainer])
+def test_workspace_rest_get_permissions(
+    workspace: Workspace,
+    user: User,
+    user_factory: UserFactory,
+    authenticated_api_client: APIClient,
+    permission: WorkspacePermission,
+):
+    workspace.set_user_permission(user, permission)
+    create_users_with_permissions(user_factory, workspace)
+    maintainer_names = [maintainer.username for maintainer in workspace.maintainers]
+    writer_names = [writer.username for writer in workspace.writers]
+    reader_names = [reader.username for reader in workspace.readers]
+
+    r = authenticated_api_client.get(f'/api/workspaces/{workspace.name}/permissions/')
+    r_json = r.json()
+
+    assert r.status_code == 200
+    assert r_json['public'] == workspace.public
+
+    for maintainer in r_json['maintainers']:
+        assert maintainer['username'] in maintainer_names
+
+    for writer in r_json['writers']:
+        assert writer['username'] in writer_names
+
+    for reader in r_json['readers']:
+        assert reader['username'] in reader_names
+
```

Comment:
Should `owner` not be part of this as well?

Reply:
Yeah, it should be. I'll add it in, but unfortunately due to the parametrization of this test, and how setup is handled, and because a workspace can only have one owner at a time, it will have to introduce some logic.

That is, unless you see a way to not have to introduce an `if/else`. Take a look once I push the fixes.